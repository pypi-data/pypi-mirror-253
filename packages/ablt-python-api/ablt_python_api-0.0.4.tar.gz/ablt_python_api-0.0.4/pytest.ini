[pytest]
markers =
    sync: marks test for synchroneous API (requests)
    asyncio: marks test for asynchroneous API (aiohttp)
addopts =  -vv -rfEx --tb=short -p no:warnings --reruns=2
testpaths =
    tests/sync
    tests/async
