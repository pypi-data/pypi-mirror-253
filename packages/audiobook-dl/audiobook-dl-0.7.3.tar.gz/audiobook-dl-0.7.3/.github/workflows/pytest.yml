name: PyTest
on:
  push:
  pull_request:
    paths:
      - '*.py'
jobs:
  tests:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2.3.4
    - uses: cachix/install-nix-action@v13
      with:
        nix_path: nixpkgs=channel:nixos-unstable
    - run: nix-shell --command "python3 -m pytest"
