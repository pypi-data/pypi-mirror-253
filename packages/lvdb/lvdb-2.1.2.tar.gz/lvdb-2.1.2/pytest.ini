[pytest]
markers =
    instance: tests for database instances
    shard: tests for database shards
    profile: profiler tests
    cache: tests for database cache