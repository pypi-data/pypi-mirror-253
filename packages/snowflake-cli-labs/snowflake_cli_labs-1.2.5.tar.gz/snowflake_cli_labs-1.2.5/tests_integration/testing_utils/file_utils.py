def replace_text_in_file(file_path: str, to_replace: str, replacement: str) -> None:
    with open(file_path, "r") as file:
        text = file.read()
    text = text.replace(to_replace, replacement)
    with open(file_path, "w") as file:
        file.write(text)
