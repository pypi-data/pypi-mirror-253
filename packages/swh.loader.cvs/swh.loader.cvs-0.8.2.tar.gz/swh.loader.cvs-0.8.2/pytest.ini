[pytest]
norecursedirs = build docs .*
asyncio_mode = strict
addopts = "--import-mode=importlib"

markers =
    fs: execute tests that write to the filesystem
