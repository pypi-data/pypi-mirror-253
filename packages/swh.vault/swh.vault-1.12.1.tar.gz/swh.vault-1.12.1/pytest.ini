[pytest]
norecursedirs = build docs .*
asyncio_mode = strict
addopts = "--import-mode=importlib"
markers =
    graph: execute tests that depend on swh-graph
