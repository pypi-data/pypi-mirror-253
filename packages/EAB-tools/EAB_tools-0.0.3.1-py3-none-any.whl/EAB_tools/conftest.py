from collections.abc import (
    Iterable,
    Iterator,
)
import itertools
import os
from pathlib import Path
from typing import (
    Any,
    Callable,
    Literal,
    Union,
)

from _pytest.fixtures import SubRequest as PytestFixtureRequest
import dateutil.tz
import matplotlib.pyplot as plt
import numpy as np
import numpy.typing as npt
import pandas as pd
from pandas._testing import makeDateIndex
import pytest

Numeric = np.number[Any]

iris_df: pd.DataFrame = pd.read_csv(Path(__file__).parent / "tests/io/data/iris.csv")


@pytest.fixture
def iris() -> pd.DataFrame:
    """The iris dataset as a pandas DataFrame."""
    return iris_df


@pytest.fixture(params=iris_df.columns)
def iris_cols(request: PytestFixtureRequest) -> pd.Series:
    """Return iris dataframe columns, one after the next"""
    return iris_df[request.param]


@pytest.fixture(
    params=[str, pytest.param(lambda col_name: pd.Index([col_name]), id="pd.Index")]
)
def iris_single_col_subset(
    iris_cols: pd.Series, request: PytestFixtureRequest
) -> Union[str, pd.Index]:
    """Return a col name as a str or pd.Index"""
    func = request.param
    col = iris_cols.name
    return func(col)


@pytest.fixture(
    params=[
        pytest.param(pd.Series([1, 2, 3] * 3, dtype="int32"), id="int32series"),
        pytest.param(
            pd.Series([None, 2.5, 3.5] * 3, dtype="float32"), id="float32series"
        ),
        pytest.param(
            pd.Series(["a", "b", "c"] * 3, dtype="category"), id="category_series"
        ),
        pytest.param(pd.Series(["d", "e", "f"] * 3), id="object_series"),
        pytest.param(pd.Series([True, False, True] * 3), id="bool_series"),
        pytest.param(
            pd.Series(pd.date_range("20130101", periods=9)), id="datetime_series"
        ),
        pytest.param(
            pd.Series(pd.date_range("20130101", periods=9, tz="US/Eastern")),
            id="datetime_tz_series",
        ),
        pytest.param(
            pd.Series(pd.timedelta_range("2000", periods=9)), id="timedelta_series"
        ),
    ]
)
def series(request: PytestFixtureRequest) -> pd.Series:
    """Return several series with unique dtypes"""
    # Fixture borrowed from pandas from
    # https://github.com/pandas-dev/pandas/blob/5b2fb093f6abd6f5022fe5459af8327c216c5808/pandas/tests/util/test_hashing.py
    return request.param


pairs = list(itertools.permutations(iris_df.columns, 2))


@pytest.fixture(params=pairs, ids=list(map(str, pairs)))
def multiindex(iris: pd.DataFrame, request: PytestFixtureRequest) -> pd.MultiIndex:
    """Return MultiIndexes created from pairs of iris cols"""
    a_col, b_col = request.param
    a, b = iris[a_col], iris[b_col]
    return pd.MultiIndex.from_arrays([a, b])


@pytest.fixture(
    params=[
        np.sin,  # any -> float
        pytest.param(lambda arr: np.exp(-arr), id="exp(-x)"),  # any -> float
        pytest.param(
            lambda x: x**2, id="lambda squared"
        ),  # int -> int and float -> float
        pytest.param(lambda arr: np.rint(arr).astype(int), id="rint"),  # any -> int
    ],
    name="func",
)
def plot_func(
    request: PytestFixtureRequest,
) -> Callable[[npt.ArrayLike], npt.NDArray[Numeric]]:
    """A variety of mathematical funcs callable on numeric numpy ndarrays"""
    return request.param


@pytest.fixture(
    params=[
        np.linspace(0, 10**-5, dtype=float),
        np.linspace(0, 499, num=500, dtype="int32"),
        np.linspace(0, 2**33, 2**10 + 1, dtype="int64"),
    ],
    ids=lambda arr: str(arr.dtype),
)
def x_values(request: PytestFixtureRequest) -> npt.NDArray[Numeric]:
    """func inputs of different dtypes"""
    return request.param


@pytest.fixture(params=["fig", "ax"])
def _fig_or_ax(request: PytestFixtureRequest) -> Literal["fig", "ax"]:
    """Either returns 'fig' or 'ax'"""
    return request.param


@pytest.fixture
def mpl_plots(
    func: Callable[[npt.ArrayLike], npt.NDArray[Numeric]],
    x_values: npt.NDArray[Numeric],
) -> Iterable[dict[str, Union[plt.Figure, plt.Axes]]]:
    """Returns dict of {fix, ax}, for various funcs and domains"""
    x = x_values
    y = func(x)
    fig, ax = plt.subplots()
    ax.plot(x, y)

    yield {"fig": fig, "ax": ax}
    plt.close(fig)


@pytest.fixture
def mpl_axes(mpl_plots: dict[str, Union[plt.Figure, plt.Axes]]) -> plt.Axes:
    """Returns a variety of `plt.Axes` objects"""
    return mpl_plots["ax"]


@pytest.fixture
def mpl_figs(mpl_plots: dict[str, Union[plt.Figure, plt.Axes]]) -> plt.Figure:
    """Returns a variety of `plt.Figure` objects"""
    return mpl_plots["fig"]


@pytest.fixture
def mpl_figs_and_axes(
    mpl_plots: dict[str, Union[plt.Figure, plt.Axes]], _fig_or_ax: Literal["fig", "ax"]
) -> Union[plt.Figure, plt.Axes]:
    """Returns either the figure or the axis of various plots"""
    return mpl_plots[_fig_or_ax]


today = pd.Timestamp.today(dateutil.tz.tzlocal())
strftime_codes = [
    "%B %d, %Y",  # June 12, 2022
    "%Y-%m-%d",  # 2022-06-12
    "%B %e, %Y",  # June 12, 2022
    "%a, %b %e",  # Sun, Jun 12
    "%e %b %Y",  # 12 Jun 2022
    "%A, %B %e, %Y",  # Sunday, June 12, 2022
    "%H:%M:%S",  # 16:51:45
    "%Y-%m-%dT%H:%M:%S.%f%z",  # 2022-06-12T16:51:45.576846-0500
    "%I:%M %p",  # 04:51 PM
]


@pytest.fixture(
    params=iter(strftime_codes),
    ids=[f"{today:{strftime}}" for strftime in strftime_codes],
)  # noqa
def strftime(request: PytestFixtureRequest) -> str:
    """Various different strftime format codes"""
    return request.param


@pytest.fixture(
    params=[
        pd.to_timedelta(0),
        pd.Timedelta(hours=15),
        pd.DataFrame(
            (pd.to_timedelta(row, unit="hours") for row in np.random.rand(10, 3) * 24),
            columns=list("ABC"),
        ),
    ]
)
def datetime_df(request: PytestFixtureRequest) -> pd.DataFrame:
    """DataFrame with datetime data, integer index, and str column names.
                   A          B          C
    0 2000-01-03 2000-01-02 2000-12-31
    1 2000-01-04 2000-01-09 2001-12-31
    2 2000-01-05 2000-01-16 2002-12-31
    3 2000-01-06 2000-01-23 2003-12-31
    4 2000-01-07 2000-01-30 2004-12-31
    5 2000-01-10 2000-02-06 2005-12-31
    6 2000-01-11 2000-02-13 2006-12-31
    7 2000-01-12 2000-02-20 2007-12-31
    8 2000-01-13 2000-02-27 2008-12-31
    9 2000-01-14 2000-03-05 2009-12-31"""
    df = pd.DataFrame(
        {
            "A": makeDateIndex(freq="b"),
            "B": makeDateIndex(freq="w"),
            "C": makeDateIndex(freq="y"),
        }
    )
    time_offset = request.param
    return df + time_offset


@pytest.fixture
def datetime_and_float_df(datetime_df: pd.DataFrame) -> pd.DataFrame:
    """datetime_df with additional columns of random positive and negative floats"""
    datetime_df[list("DE")] = np.random.uniform(-1, 1, (10, 2))
    return datetime_df


@pytest.fixture(autouse=True)
def _docstring_tmp_path(request: PytestFixtureRequest) -> Iterator[None]:
    # Almost completely adapted from a kind soul at https://stackoverflow.com/a/46991331
    # Trigger ONLY for the doctests.
    doctest_plugin = request.config.pluginmanager.getplugin("doctest")
    if isinstance(request.node, doctest_plugin.DoctestItem):
        # Get the fixture dynamically by its name.
        tmp_path: Path = request.getfixturevalue("tmp_path")
        # Chdir only for the duration of the test.
        og_dir = os.getcwd()
        try:
            os.chdir(tmp_path)
            yield
        finally:
            os.chdir(og_dir)
    else:
        # For normal tests, we have to yield, since this is a yield-fixture.
        yield
