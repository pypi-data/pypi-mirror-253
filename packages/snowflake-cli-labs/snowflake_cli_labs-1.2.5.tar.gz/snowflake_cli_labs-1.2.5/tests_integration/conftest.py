from __future__ import annotations

import functools
import json
import tempfile
import shutil
import os

import pytest
from dataclasses import dataclass
from pathlib import Path
from snowcli.app.cli_app import app
from typer import Typer
from typer.testing import CliRunner
from typing import List, Dict, Any, Optional

TEST_DIR = Path(__file__).parent


@dataclass
class CommandResult:
    exit_code: int
    json: Optional[List[Dict[str, Any]] | Dict[str, Any]] = None
    output: Optional[str] = None


@pytest.fixture(scope="session")
def test_snowcli_config():
    test_config_name = "connection_configs.toml"
    test_config = TEST_DIR / "config" / test_config_name
    with tempfile.TemporaryDirectory() as td:
        test_config_path = os.path.join(td, test_config_name)
        shutil.copyfile(test_config, test_config_path)
        yield test_config_path


@pytest.fixture(scope="session")
def test_root_path():
    return TEST_DIR


class SnowCLIRunner(CliRunner):
    def __init__(self, app: Typer, test_snowcli_config: str):
        super().__init__()
        self.app = app
        self.test_snowcli_config = test_snowcli_config

    @functools.wraps(CliRunner.invoke)
    def _invoke(self, *a, **kw):
        kw.update(catch_exceptions=False)
        return super().invoke(self.app, *a, **kw)

    def invoke_with_config(self, *args, **kwargs) -> CommandResult:
        result = self._invoke(
            ["--config-file", self.test_snowcli_config, *args[0]],
            **kwargs,
        )
        return CommandResult(result.exit_code, output=result.output)

    def invoke_integration(self, *args, **kwargs) -> CommandResult:
        result = self._invoke(
            [
                "--config-file",
                self.test_snowcli_config,
                *args[0],
                "--format",
                "JSON",
                "-c",
                "integration",
            ],
            **kwargs,
        )
        if result.output == "" or result.output.strip() == "Done":
            return CommandResult(result.exit_code, json=[])
        return CommandResult(result.exit_code, json.loads(result.output))

    def invoke_integration_without_format(self, *args, **kwargs) -> CommandResult:
        result = self._invoke(
            [
                "--config-file",
                self.test_snowcli_config,
                *args[0],
                "-c",
                "integration",
            ],
            **kwargs,
        )
        return CommandResult(result.exit_code, output=result.output)


@pytest.fixture
def runner(test_snowcli_config):
    return SnowCLIRunner(app, test_snowcli_config)
