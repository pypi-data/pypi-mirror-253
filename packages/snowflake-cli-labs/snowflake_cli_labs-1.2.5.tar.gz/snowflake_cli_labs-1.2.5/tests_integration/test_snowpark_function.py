from __future__ import annotations

import pytest

from tests_integration.snowflake_connector import test_database, snowflake_session
from tests_integration.testing_utils.snowpark_utils import (
    TestType,
    SnowparkTestSetup,
    SnowparkTestSteps,
)
from tests_integration.testing_utils.sql_utils import sql_test_helper
from tests_integration.testing_utils.naming_utils import object_name_provider
from tests_integration.testing_utils.working_directory_utils import (
    temporary_working_directory,
)
from tests_integration.conftest import SnowCLIRunner


@pytest.mark.integration
def test_snowpark_function_flow(_test_steps):
    _test_steps.assert_that_no_entities_are_in_snowflake()
    _test_steps.assert_that_no_files_are_staged_in_test_db()

    _test_steps.snowpark_list_should_return_no_data()

    _test_steps.snowpark_init_should_initialize_files_with_default_content()
    _test_steps.snowpark_package_should_zip_files()

    function_name = _test_steps.snowpark_create_should_finish_successfully()
    _test_steps.assert_that_only_these_entities_are_in_snowflake(
        f"{function_name}() RETURN VARCHAR"
    )
    _test_steps.assert_that_only_these_files_are_staged_in_test_db(
        f"deployments/{function_name}/app.zip"
    )

    _test_steps.snowpark_list_should_return_entity_at_first_place(
        entity_name=function_name,
        arguments="()",
        result_type="VARCHAR",
    )

    _test_steps.snowpark_describe_should_return_entity_description(
        entity_name=function_name,
        arguments="()",
    )

    _test_steps.snowpark_execute_should_return_expected_value(
        entity_name=function_name,
        arguments="()",
        expected_value="Hello World!",
    )

    _test_steps.snowpark_update_should_not_replace_if_the_signature_does_not_change(
        function_name
    )
    _test_steps.snowpark_execute_should_return_expected_value(
        entity_name=function_name, arguments="()", expected_value="Hello Snowflakes!"
    )

    _test_steps.snowpark_update_should_finish_successfully(function_name)
    _test_steps.assert_that_only_these_entities_are_in_snowflake(
        f"{function_name}() RETURN NUMBER"
    )
    _test_steps.assert_that_only_these_files_are_staged_in_test_db(
        f"deployments/{function_name}/app.zip"
    )

    _test_steps.snowpark_list_should_return_entity_at_first_place(
        entity_name=function_name,
        arguments="()",
        result_type="NUMBER",
    )

    _test_steps.snowpark_execute_should_return_expected_value(
        entity_name=function_name,
        arguments="()",
        expected_value=1,
    )

    _test_steps.snowpark_drop_should_finish_successfully(
        entity_name=function_name,
        arguments="()",
    )
    _test_steps.assert_that_no_entities_are_in_snowflake()
    _test_steps.assert_that_only_these_files_are_staged_in_test_db(
        f"deployments/{function_name}/app.zip"
    )

    _test_steps.snowpark_list_should_return_no_data()


@pytest.fixture
def _test_setup(
    runner,
    snowflake_session,
    test_database,
    sql_test_helper,
    object_name_provider,
    temporary_working_directory,
    snapshot,
):
    snowpark_function_test_setup = SnowparkTestSetup(
        runner=runner,
        snowflake_session=snowflake_session,
        sql_test_helper=sql_test_helper,
        object_name_provider=object_name_provider,
        snapshot=snapshot,
        test_database=test_database,
        test_type=TestType.FUNCTION,
    )
    yield snowpark_function_test_setup
    snowpark_function_test_setup.clean_after_test_case()


@pytest.fixture
def _test_steps(_test_setup):
    yield SnowparkTestSteps(_test_setup, TestType.FUNCTION)
