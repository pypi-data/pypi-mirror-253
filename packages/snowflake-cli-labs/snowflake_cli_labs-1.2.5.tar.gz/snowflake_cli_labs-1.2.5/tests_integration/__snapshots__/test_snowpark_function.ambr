# serializer version: 1
# name: test_snowpark_function_flow[.gitignore]
  '''
  .packages/
  .venv/
  app.zip
  __pycache__
  
  '''
# ---
# name: test_snowpark_function_flow[app.py]
  '''
  from __future__ import annotations
  
  import sys
  
  
  def hello() -> str:
      return "Hello World!"
  
  
  # For local debugging. Be aware you may need to type-convert arguments if
  # you add input parameters
  if __name__ == "__main__":
      if len(sys.argv) > 1:
          print(hello(sys.argv[1:]))  # type: ignore
      else:
          print(hello())  # type: ignore
  
  '''
# ---
# name: test_snowpark_function_flow[config.toml]
  '''
  # Can create multiple --environments (default 'dev')
  
  [connections]
  [connections.dev]
  account = "<ACCOUNT_NAME>"
  database = "<DATABASE_NAME>"
  schema = "<SCHEMA_NAME>"
  warehouse = "<WAREHOUSE_NAME>"
  role = "<ROLE_NAME>"
  
  '''
# ---
# name: test_snowpark_function_flow[requirements.txt]
  '''
  snowflake-snowpark-python
  
  '''
# ---
